src/platform/graphics/drm_formats.cpp

```diff
@@ -442,68 +444,60 @@ constexpr auto maybe_info_for_format(uint32_t fourcc_format) -> mg::DRMFormat::F
     }
 
 }
-
-constexpr auto info_for_format(uint32_t fourcc_format) -> mg::DRMFormat::FormatInfo const&
-{
-    auto const info = maybe_info_for_format(fourcc_format);
-
-    if (info)
-    {
-        return *info;
-    }
-    BOOST_THROW_EXCEPTION((
-        std::runtime_error{
-            std::string{"Unsupported DRM format: "} + mg::drm_format_to_string(fourcc_format)}));
 }
 
+auto mg::DRMFormat::name() const -> const char*
+{
+    return drm_format_to_string(fourcc);
 }
 
-mg::DRMFormat::DRMFormat(uint32_t fourcc_format)
-    : info{&info_for_format(fourcc_format)}
+auto mg::DRMFormat::Info::opaque_equivalent() const -> std::optional<DRMFormat>
 {
+    return info->opaque_equivalent.transform([](auto const& fourcc) { return DRMFormat{fourcc};});
 }
 
-auto mg::DRMFormat::name() const -> const char*
+auto mg::DRMFormat::Info::alpha_equivalent() const -> std::optional<DRMFormat>
 {
-    return drm_format_to_string(info->format);
+    return info->alpha_equivalent.transform([](auto const& fourcc) { return DRMFormat{fourcc};});
 }
 
-auto mg::DRMFormat::opaque_equivalent() const -> const std::optional<DRMFormat>
+bool mg::DRMFormat::Info::has_alpha() const
 {
-    if (info->opaque_equivalent != DRM_FORMAT_INVALID)
-    {
-        return DRMFormat{info->opaque_equivalent};
-    }
-    return {};
+    return info->has_alpha;
 }
 
-auto mg::DRMFormat::alpha_equivalent() const -> const std::optional<DRMFormat>
+auto mg::DRMFormat::Info::components() const -> std::optional<Info::RGBComponentInfo> const&
 {
-    if (info->alpha_equivalent != DRM_FORMAT_INVALID)
-    {
-        return DRMFormat{info->alpha_equivalent};
-    }
-    return {};
+    return info->components;
 }
 
-bool mg::DRMFormat::has_alpha() const
+mg::DRMFormat::Info::Info(FormatInfo const* info)
+    : info{info}
 {
-    return info->has_alpha;
 }
 
-auto mg::DRMFormat::components() const -> std::optional<RGBComponentInfo> const&
+auto mg::DRMFormat::info() const -> std::optional<Info const>
 {
-    return info->components;
+    if (auto info = maybe_info_for_format(fourcc))
+    {
+        return Info(info);
+    }
+    /* TODO: This could fire quite frequently.
+     * Ideally we would have a rate-limited (or once-per-format) logger, but for now, just
+     * throw it in the debug stream.
+     */
+    mir::log_debug("Detailed info for format %s missing; please report so this can be added", name());
```

Comment:
There's a lot of TODOs being added. In `linux_dmabuf.cpp` we say something similar, but in that context we don't log. I think we need a better approach to reporting this. BTW "rate-limiting" doesn't sound right either.

With this approach, I think "debug" logging requesting reporting is addressing the wrong audience. "debug" and "trace" are for those familiar with the code. "Warning" might be better.